controller/controller.go

```diff
@@ -45,24 +61,27 @@ func DefaultConfig() Config {
 type Controller interface {
 	// Submit schedules a group on the configured fleet cluster. This is done by
 	// setting the state of the units in the group to loaded.
-	Submit(req Request) (*task.Task, error)
+	Submit(req Request, closer <-chan struct{}) (*task.Task, error)
 
 	// Start starts a group on the configured fleet cluster. This is done by
 	// setting the state of the units in the group to launched.
-	Start(req Request) (*task.Task, error)
+	Start(req Request, closer <-chan struct{}) (*task.Task, error)
 
 	// Stop stops a group on the configured fleet cluster. This is done by
 	// setting the state of the units in the group to loaded.
-	Stop(req Request) (*task.Task, error)
+	Stop(req Request, closer <-chan struct{}) (*task.Task, error)
 
 	// Destroy delets a group on the configured fleet cluster. This is done by
 	// setting the state of the units in the group to inactive.
-	Destroy(req Request) (*task.Task, error)
+	Destroy(req Request, closer <-chan struct{}) (*task.Task, error)
```

Comment:
Don't know. The context is for different things. Some of these makes no sense here. It is just that we would use one fraction of a feature set of something. So the fit of this asset would not be that good.